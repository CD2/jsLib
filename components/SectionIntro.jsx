import React from 'react'
import PropTypes from 'prop-types'
import decorate from 'utils/decorate'
import windowStore from 'stores/window'
import { styled, t } from 'utils/theme'

export class SectionIntro extends React.Component {

  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    heading: PropTypes.number,
    light: PropTypes.bool,
    title: PropTypes.string,
  }

  render() {
    const { className, children, title, heading } = this.props
    return (
      <div className={className}>
        {heading === 2 ?
          <h2 className="page-intro__heading">{title}</h2> :
          heading === 3 ?
            <h3 className="page-intro__heading">{title}</h3> :
            <h1 className="page-intro__heading">{title}</h1>}
        {
          children &&
          <div className="intro-text">
            {children}
          </div>
        }
      </div>
    )
  }

}

export default decorate(
  styled`
    .modal & { padding-top: 0; }
    ${({ children, theme }) => {
    if(children){
      return `padding-bottom: ${theme.gutterHeight.value / 16}em;`  
    }
  }}
    .intro-text {
      ${({ light, theme }) => {
    const color = light ? theme.background : theme.lightText
    return `color: ${color};`
  }}
    }
    .page-intro__heading {
      color: ${t(`headingText`)};
      margin: 0;
      ${windowStore.isSmall ? `font-size: 2em;` : ``};
      ${({ light }) => {
    if(light) {
      return `color: white;`
    }
  }}
    }
    .intro-text h3 {
      margin: 0 0 ${props => props.gutterHeight || props.theme.gutterHeight.value}px;;
      max-width: 800px;
      color: ${t(`lightText`)};
      ${({ light }) => {
    if (light) {
      return `color: white;`
    }
  }}
      font-weight: 500;
    }
    p {
      margin: 0 0 ${props => props.gutterHeight || props.theme.gutterHeight.value}px;
      max-width: 800px;
      &:last-child {
        margin-bottom: 0;
      }
    }
    input {
      margin-top: ${t(`spacing.medium`, t=>t/16)}em;
    }
    + {
      p, h1, h2, h3, h4 {
        margin-top: 0;
      }
    }
    .btn {
      margin-top: 16px;
    }
  `,
  SectionIntro
)
