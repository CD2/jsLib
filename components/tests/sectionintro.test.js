/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { SectionIntro } from '../SectionIntro'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const wrapper = shallow(<SectionIntro />)
it(`snapshot`, ()=>expect(toJson(wrapper)).toMatchSnapshot())
