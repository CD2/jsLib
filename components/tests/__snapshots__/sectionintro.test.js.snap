// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionIntro /> snapshot 1`] = `
<div
  style={Object {}}
>
  <h1
    className="page-intro__heading"
  />
</div>
`;

exports[`snapshot 1`] = `
<div
  style={Object {}}
>
  <h1
    className="page-intro__heading"
  />
</div>
`;
